File: android/src/main/java/expo/modules/fetcher/types/FetchResponse.kt
package expo.modules.fetcher.types

import expo.modules.kotlin.records.Field
import expo.modules.kotlin.records.Record

class FetchResponse : Record {
  @Field
  var status: Int = 0

  @Field
  var statusText: String = ""

  @Field
  var headers: List<String> = emptyList()

  @Field
  var bodyBase64: String = ""

  @Field
  var url: String = ""

  @Field
  var ok: Boolean = false

  @Field
  var redirected: Boolean = false

  @Field
  var type: String = "basic"
}

File: ios/Types/FetchResponse.swift
import ExpoModulesCore

struct FetchResponse: Record {
  @Field var status: Int
  @Field var statusText: String
  @Field var headers: [String]
  @Field var bodyBase64: String
  @Field var url: String
  @Field var ok: Bool
  @Field var redirected: Bool
  @Field var type: String
}
